Convert.py

```diff
@@ -82,63 +92,141 @@ def signy(y):
-        gen_pt.push_back(pt)
-        gen_eta.push_back(eta)
-        gen_phi.push_back(phi)
-        gen_energy.push_back(math.sqrt(g.core.p4.mass**2+g.core.p4.px**2+g.core.p4.py**2+g.core.p4.pz**2))
-
-        if math.fabs(tlv.E()-math.sqrt(g.core.p4.mass**2+g.core.p4.px**2+g.core.p4.py**2+g.core.p4.pz**2))>0.01 and g.core.status==1:
-            print '=======================etlv  ',tlv.E(),'    ',math.sqrt(g.core.p4.mass**2+g.core.p4.px**2+g.core.p4.py**2+g.core.p4.pz**2),'  eta  ',eta,'   phi   ',phi,'  x  ',g.core.p4.px,'  y  ',g.core.p4.py,'  z  ',g.core.p4.pz
+    ev_num = event.GetEvent()
+    E = .0
+    Ebench = .0
+    Eem = 0
+    Ehad = 0
+    EemFirst = 0
+    EhadLast = 0
+    if event.GetBranchStatus("GenParticles"):
+        for g in event.GenParticles:
+            position = r.TVector3(g.core.p4.px,g.core.p4.py,g.core.p4.pz)
+            
+            pt=math.sqrt(g.core.p4.px**2+g.core.p4.py**2)
+            eta=position.Eta()
+            phi=position.Phi()
+            
+            tlv=r.TLorentzVector()
+            tlv.SetPtEtaPhiM(pt,eta,phi,g.core.p4.mass)
+            gen_pt.push_back(pt)
+            gen_eta.push_back(eta)
+            gen_phi.push_back(phi)
+            gen_energy.push_back(math.sqrt(g.core.p4.mass**2+g.core.p4.px**2+g.core.p4.py**2+g.core.p4.pz**2))
+
+            if math.fabs(tlv.E()-math.sqrt(g.core.p4.mass**2+g.core.p4.px**2+g.core.p4.py**2+g.core.p4.pz**2))>0.01 and g.core.status==1:
+                print '=======================etlv  ',tlv.E(),'    ',math.sqrt(g.core.p4.mass**2+g.core.p4.px**2+g.core.p4.py**2+g.core.p4.pz**2),'  eta  ',eta,'   phi   ',phi,'  x  ',g.core.p4.px,'  y  ',g.core.p4.py,'  z  ',g.core.p4.pz
         #if g.core.status==1:
             #print 'etlv  ',tlv.E(),'    ',math.sqrt(g.core.p4.mass**2+g.core.p4.px**2+g.core.p4.py**2+g.core.p4.pz**2),'  eta  ',eta,'   phi   ',phi,'  x  ',g.core.p4.px,'  y  ',g.core.p4.py,'  z  ',g.core.p4.pz
-        gen_pdgid.push_back(g.core.pdgId)
-        gen_status.push_back(g.core.status)
+            gen_pdgid.push_back(g.core.pdgId)
+            gen_status.push_back(g.core.status)
 
     for c in event.HCalBarrelCellPositions:
+        hcalBarrel_decoder.setValue(c.core.cellId)
         position = r.TVector3(c.position.x,c.position.y,c.position.z)
         rec_ene.push_back(c.core.energy)
         rec_eta.push_back(position.Eta())
         rec_phi.push_back(position.Phi())
         rec_pt.push_back(c.core.energy*position.Unit().Perp())
-        rec_layer.push_back(layerID(c.position.x,c.position.y))
+        rec_layer.push_back(hcalBarrel_decoder["layer"] + lastECalBarrelLayer + 1)
         rec_x.push_back(c.position.x/10.)
         rec_y.push_back(c.position.y/10.)
         rec_z.push_back(c.position.z/10.)
-        #print '  new eta HCAL  ',position.Eta(), ' pt  ',c.core.energy*position.Unit().Perp()
+        rec_detid.push_back(systemID(c.core.cellId))
+        if hcalBarrel_decoder["layer"] == 0:
```

Comment:
is 0 really the last layer? this variable shoudl be EhadFirst I think

Reply:
0 is always the first layer, I don't think we need another variable for it